framework/source/class/qx/ui/window/Window.js

```diff
@@ -929,6 +1055,40 @@ qx.Class.define("qx.ui.window.Window",
       }
     },
 
+    // overridden
+    _applyCenterWhen : function(value, old)
+    {
+      var             parent;
+
+      // Remove prior listener for centering on appear
+      if (this.__centeringAppearId !== null) {
+        this.removeListenerById(this.__centeringAppearId);
+        this.__centeringAppearId = null;
+      }
```

Comment:
I think all this logic could be omitted including the dynamic removal and re-assignment of the listeners. I'll add the listeners for resize and appear anyway and ask in the listener if the corresponding property is true and center or not. I don't think it's worth all that listener tracking.